Plugins/PluginsShared/PluginError.swift

```diff
@@ -0,0 +1,82 @@
+import PackagePlugin
+import Foundation
+
+enum PluginError: Swift.Error, CustomStringConvertible, LocalizedError {
+    case incompatibleTarget(targetName: String)
+    case noTargetsFoundForCommandPlugin
+    case fileErrors([FileError], targetName: String)
+
+    var description: String {
+        switch self {
+        case .incompatibleTarget(let targetName):
+            return "Incompatible target called '\(targetName)'. Only Swift source targets can be used with the Swift OpenAPI generator plugin."
+        case .noTargetsFoundForCommandPlugin:
+            return "None of the targets include valid OpenAPI spec files. Please make sure at least one of your targets has any valid OpenAPI spec files before triggering this command plugin. See documentation for details."
+        case .fileErrors(let errors, let targetName):
+            return "Found file errors in target called '\(targetName)': \(errors.description)"
+        }
+    }
+
+    var errorDescription: String? {
+        description
+    }
+}
+
+extension [PluginError]: Swift.Error, CustomStringConvertible, LocalizedError {
+    public var errorDescription: String? {
+        description
+    }
+}
+
+struct FileError: Swift.Error, CustomStringConvertible, LocalizedError {
+
+    enum Kind: CaseIterable {
+        case config
+        case document
+    }
+
+    enum Issue {
+        case notFound
+        case multiFound(files: [Path])
```

Comment:
```suggestion
        case multipleFilesFound(files: [Path])
```

While we're changing this, we can make it clearer.